training/views.py

```diff
@@ -106,131 +139,134 @@ def trainings_for(trainings, year, month, day):
-        training_identifier="test"
-    )  # Exclude the 'test' group from showing up in calendar
-    approved_trainings = [x for x in trainings if x.processed == "AP"]
-    approved = len(approved_trainings)
-    if len(approved_trainings) > 0:
-        start = min([x.start for x in approved_trainings])
-        end = max([x.end for x in approved_trainings])
-    else:
-        start = date.today()
-        end = date.today()
-
-    years = list(range(start.year, end.year + 1))[::-1]
-
-    months = [
-        "January",
-        "February",
-        "March",
-        "April",
-        "May",
-        "June",
-        "July",
-        "August",
-        "September",
-        "October",
-        "November",
-        "December",
-    ]
-
-    max_value = 0
-
-    days = {}
-    for year in years:
-        for idx, month in enumerate(months):
-            if year not in days:
-                days[year] = {}
-
-            days_au = calendar.monthcalendar(year, idx + 1)
-            days_fix = []
-            for row in days_au:
-                new_row = [
-                    (x, trainings_for(approved_trainings, year, idx + 1, x))
-                    for x in row
-                ]
-                m = max([x[1] for x in new_row])
-                if m > max_value:
-                    max_value = m
-                days_fix.append(new_row)
-
-            days[year][month] = days_fix
-
+    """Display scheduled events in an interactive calendar view."""
+    approved_trainings = (
+        Training.objects.all()
+        .exclude(training_identifier="test")
+        .filter(processed="AP")
+        .order_by('start')
+    )
     return render(
         request,
         "training/calendar.html",
         {
-            "trainings": approved,
-            "start": start,
-            "end": end,
-            "years": years,
-            "months": months,
-            "days": days,
-            "max_value": max_value,
-            "settings": settings,
+            "events": approved_trainings,
+            "admin_user": request.user.is_staff,
+            "n_events": approved_trainings.count()
         },
     )
 
 
 def stats(request):
-    # Exclude the 'test' group from showing up in calendar
-    trainings = Training.objects.all().exclude(training_identifier="test")
-    approved = len([x for x in trainings if x.processed == "AP"])
-    waiting = len([x for x in trainings if x.processed == "UN"])
-    days = sum([(x.end - x.start).days for x in trainings])
-    students = sum([x.attendance for x in trainings])
-
-    if len(trainings) > 0:
-        current_trainings = len([x for x in trainings if x.start <= date.today() <= x.end])
-        earliest = min([x.start for x in trainings])
-    else:
-        current_trainings = 0
-        earliest = date.today()
+    trainings = Training.objects.exclude(
+        training_identifier="test"
+    )  # Exclude the 'test' group from showing up in calendar
```

Comment:
All of this reformatting is really making review difficult :/

Reply:
Sorry about that. It more efficient (and mostly clearer) querying with the ORM though!

Reply:
No definitely, it's an improvement (especially at scale, which, <1k rows probably isn't.)